pkgs/test_core/lib/src/util/io.dart

```diff
@@ -226,3 +227,17 @@ Future<Uri> getRemoteDebuggerUrl(Uri base) async {
     return base;
   }
 }
+
+extension RetryDelete on FileSystemEntity {
+  Future<void> deleteWithRetry() async {
+    for (var i = 0; i < 3; i++) {
```

Comment:
nit: consider a while(true) instead and use the return/rethrow to exit (increment i inside the loop). That would keep the logic for the number of retries in one place instead of 2, slightly less error prone imo.

Reply:
Done.